static/js/services/lineage-model-generator.js

```diff
@@ -63,13 +63,37 @@ angular.module('inboxServices').factory('LineageModelGenerator',
         });
     };
 
+    var mergeParents = function(doc, lineage) {
+      var current = doc;
+      lineage.forEach(function(hydrated) {
+        if (!current) {
+          return;
+        }
+        if (hydrated) {
+          current.parent = hydrated;
+        }
+        current = current.parent;
+      });
+      return doc;
+    };
+
     return {
-      contact: function(id) {
+
+      /**
+       * Fetch a contact and its lineage by the given uuid. Returns a
+       * contact model, or if options.merge is true the doc with the
+       * lineage inline.
+       */
+      contact: function(id, options) {
+        options = options || {};
         return get(id).then(function(docs) {
           // the first row is the contact
           var doc = docs.shift();
            // everything else is the lineage
           return hydrate(docs).then(function(lineage) {
+            if (options.merge) {
+              return mergeParents(doc, lineage);
```

Comment:
this returns just `doc`, whereas the non-merged option on the same function returns `{_id: id, doc: doc}`. That's a really sneaky return value, can you make them both return the same format (either one)

Reply:
Done.